runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java

```diff
@@ -485,7 +555,7 @@ private HiscoreSkill find(int index)
 				return BOUNTY_HUNTER_HUNTER;
 		}
 
-		return null;
+		return BOSSES.get(index - (SKILLS.size() + 7));
```

Comment:
this is still awful; I have no idea what this SKILLS+7 is for

Reply:
the skillLabels for boss kcs start at position `SKILLS.size() + 7`  in the skillLabels List so I need to lower the index by that amount to retrieve the respective boss in the BOSSES List